libtenzir/src/detail/posix.cpp

```diff
@@ -21,24 +20,32 @@
 #include <sys/types.h>
 #include <sys/un.h>
 
-#include <cerrno>
 #include <cstring>
+#include <errno.h>
 #include <fcntl.h>
 #include <limits>
-#include <stdexcept>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
 namespace tenzir::detail {
 
 auto describe_errno(int err) -> std::string {
-  auto result = std::string(256, '\0');
-  if (strerror_r(err, result.data(), result.size()) != 0) {
-    return fmt::format("<errno = {}>", err);
-  }
-  result.erase(std::find(result.begin(), result.end(), '\0'), result.end());
-  return result;
+  auto buffer = std::string(256, '\0');
+  auto result = ::strerror_r(err, buffer.data(), buffer.size());
+  return [&](auto result) {
+    if constexpr (std::same_as<decltype(result), int>) {
+      if (result != 0) {
+        return fmt::format("<errno = {}>", err);
+      }
+      buffer.erase(std::ranges::find(buffer, '\0'), buffer.end());
+      return buffer;
+    } else {
+      static_assert(std::same_as<decltype(result), char*>);
+      TENZIR_ASSERT(result);
+      return std::string(result);
+    }
+  }(result);
```

Comment:
@dominiklohmann Please take a look at this once more and resolve if it's good to go. 

I had to change this again because the `syserr_list` approach is not portable, and `strerror_r` returns a `char*` on Linux, which passes type checking with `!= 0`, but has totally wrong semantics... Seems like getting textual descriptions of errors is still a hard problem in 2024.

Reply:
Oh wow, the promotion from the 0 literal to nullptr striking again.

I did not test this locally, but the code makes sense to me. It's easy enough to follow. Feel free to resolve.